lib/trino-parquet/src/main/java/io/trino/parquet/reader/PageReader.java

```diff
@@ -103,6 +103,9 @@ public DataPage readPage()
         dataPageReadCount++;
         try {
             if (compressedPage instanceof DataPageV1 dataPageV1) {
+                if (!arePagesCompressed()) {
+                    return dataPageV1;
```

Comment:
is `dataPageV1` reused in any way?

Reply:
Existing code is also logically the same, `decompress` will just give back the input slice without modifications for uncompressed case.